### ISSUE TITLE
Algorithm for asserting that a TD is a valid TM instance

### ISSUE THREAD

**Issue by JKRhb** [w3c/wot-thing-description#1797]
In the TD specification, the assertion [tm-derivation-validity](https://w3c.github.io/wot-thing-description/#tm-derivation-validity) prescribes that a Thing Description "instantiated by relying on a Thing Model [...] SHOULD be valid according to that Thing Model". 

Although the process to ensure such validity presented by @egekorkan during the developer meetup two weeks ago looks very intuitive, I think it would be nice to have some kind of algorithm for formally performing this kind of validation in TD 2.0.

Besides making it easier for developers to include a validation process in their implementations, this would also make it more clear what the actual criteria for this kind of validity are, which could then be guaranteed more easily during the TM to TD derivation process.

**Comment by egekorkan** [comment]
I agree that we need this to be pseudo-code like algorithm. In the meantime, you can check how this validation is done in playground at https://github.com/thingweb/thingweb-playground/blob/master/packages/core/shared.js . The functions in question are checkLinkedAffordances and checkLinkedStructure . node-wot is also used for fetching all linked TMs.
